runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -921,20 +925,494 @@ static void VMwrtbarEvaluator(TR::Node *node, TR::Register *srcReg, TR::Register
+   startLabel->setStartInternalControlFlow();
+   endLabel->setEndInternalControlFlow();
+
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, startLabel);
+
+   uint8_t numOfConditions = getNumOfConditionsForReportFieldAccess(node, isResolved, isWrite,isInstanceField, cg);
+   numOfConditions++;
+   TR::RegisterDependencyConditions  *deps =  new (cg->trHeapMemory())TR::RegisterDependencyConditions(numOfConditions, numOfConditions, cg->trMemory());
+
+   TR_PPCOutOfLineCodeSection *generateReportOOL = new (cg->trHeapMemory()) TR_PPCOutOfLineCodeSection(fieldReportLabel, endLabel, cg);
+   cg->getPPCOutOfLineCodeSectionList().push_front(generateReportOOL);
+   
+   // Load fieldClass
+   if (isInstanceField)
+      {
+      generateLoadJ9Class(node, fieldClassReg, sideEffectRegister, cg);
+      }
+   else if (isResolved)
+      {
+      /* During Non-AOT compilation the fieldClass has been populated inside the dataSnippet during compilation. 
+      During AOT compilation the fieldClass must be loaded from the snippet. The fieldClass in an AOT body is invalid. */
+      if (isAOTCompile)
+         {
+         // Load FieldClass from snippet
+         generateTrg1MemInstruction(cg,TR::InstOpCode::Op_load, node, fieldClassReg,
+                           new (cg->trHeapMemory()) TR::MemoryReference(dataSnippetRegister, offsetof(J9JITWatchedStaticFieldData, fieldClass),
+                           TR::Compiler->om.sizeofReferenceAddress(), cg));
+         }
+      else
+         {
+         J9Class * fieldClass = static_cast<TR::J9PPCWatchedStaticFieldSnippet *>(dataSnippet)->getFieldClass();
+         loadAddressConstant(cg, node, reinterpret_cast<uintptrj_t>(fieldClass), fieldClassReg);
+         }
+      }
+   else
+      {
+      // Unresolved 
+      if (isWrite)
+         {
+         generateTrg1MemInstruction(cg,TR::InstOpCode::Op_load, node, fieldClassReg,
+               new (cg->trHeapMemory()) TR::MemoryReference(sideEffectRegister, fej9->getOffsetOfClassFromJavaLangClassField(), TR::Compiler->om.sizeofReferenceAddress(), cg));
+         }
+      else
+         {
+         generateTrg1Src2Instruction(cg, TR::InstOpCode::OR, node, fieldClassReg, sideEffectRegister, sideEffectRegister);
+         }
+      } 
+
+   classFlagsMemRef = new (cg->trHeapMemory()) TR::MemoryReference(fieldClassReg, static_cast<uintptrj_t>(fej9->getOffsetOfClassFlags()),TR::Compiler->om.sizeofReferenceAddress(), cg);
+   
+   generateTrg1MemInstruction(cg,TR::InstOpCode::Op_load, node, scratchReg, classFlagsMemRef);
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::andi_r, node, scratchReg, scratchReg, cndReg, J9ClassHasWatchedFields);
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::bne, node, fieldReportLabel, cndReg);
+
+   deps->addPreCondition(sideEffectRegister, TR::RealRegister::NoReg);
+   deps->addPostCondition(sideEffectRegister, TR::RealRegister::NoReg);
+
+   generateReportOOL->swapInstructionListsWithCompilation();
+      {
```

Comment:
Same as above. What is the purpose of these braces? I think we can remove them.

Reply:
As above - bound by precedence.

Reply:
See https://github.com/eclipse/openj9/pull/6234/files#r296764337

Reply:
Changes were made here:
https://github.com/eclipse/openj9/pull/6234/files#diff-efcdf3eda93e6ae70d4c46dd0bf9b5acR1330

@dchopra001 going to resolve this one on your behalf.